File: src/main/java/tk/hiddenname/smarthome/entity/signal/SignalType.java
package tk.hiddenname.smarthome.entity.signal;

import tk.hiddenname.smarthome.exception.SignalTypeNotFoundException;

public enum SignalType {
    DIGITAL,
    PWM;

    public static SignalType getSignalType(String type) {
        try {
            return SignalType.valueOf(type.toUpperCase());
        } catch (IllegalArgumentException e) {
            throw new SignalTypeNotFoundException(type);
        }
    }
}

File: src/main/java/tk/hiddenname/smarthome/entity/signal/SignalType.kt
package tk.hiddenname.smarthome.entity.signal

import tk.hiddenname.smarthome.exception.SignalTypeNotFoundException

enum class SignalType {
    DIGITAL,
    PWM
}

fun getSignalType(signalTypeName: String): SignalType {
    try {
        return SignalType.valueOf(signalTypeName.toUpperCase())
    } catch (e: IllegalArgumentException) {
        throw SignalTypeNotFoundException(signalTypeName)
    }
}